gpt4all-bindings/python/gpt4all/_pyllmodel.py

```diff
@@ -32,7 +32,7 @@
 # Detect Rosetta 2
 if platform.system() == "Darwin" and platform.processor() == "i386":
     if subprocess.run(
-        "sysctl -n sysctl.proc_translated".split(), check=True, capture_output=True, text=True,
+        "sysctl -n sysctl.proc_translated".split(), capture_output=True, text=True,
     ).stdout.strip() == "1":
```

Comment:
I would prefer if we put this in a function (prefixed with an underscore, e.g. `def _check_rosetta():`) so we can use local variables. Then we can do something a little more robust:
```python
    p = subprocess.run("sysctl -n sysctl.proc_translated".split(), capture_output=True, text=True)
    if p.returncode == 0 and p.stdout.strip() == "1":
        # ...
```

We don't want to throw an unhandled exception on Intel macs, but it's always good practice to check the return code of an executed process.